### ISSUE TITLE
Cannot build android production due to "Execution failed for task ':opentok-react-native:verifyReleaseResources'."

### ISSUE THREAD

**Issue by nuttikung** [opentok/opentok-react-native#151]
# Bug Report

**Current behavior**
> I have tried to compile and build for production apk and I got an error shown as
"FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':opentok-react-native:verifyReleaseResources'.
> com.android.ide.common.process.ProcessException: Failed to execute aapt "

**Steps to reproduce**
> 
1. bundle js  via react native command.
2. cd android && ./gradlew assembleRelease
3. the error will be shown.

**Example Project**
> -

**What is the current *bug* behavior?**
> It's shown as BUILD FAILED with error as I mention above.

**What is the expected *correct* behavior?**
> It should show as BUILD SUCCESS and show up for apk file.

**Relevant logs and/or screenshots**
> 
<img width="767" alt="screen shot 2018-08-15 at 16 52 48" src="https://user-images.githubusercontent.com/8456620/44142822-ad0f38fc-a0ab-11e8-88c8-5f111d07382e.png">

**Comment by msach22** [comment]
@nuttikung Apologies for the delay in checking this out, are you seeing this issue with the latest release (0.8.3)?

**Comment by msach22** [comment]
@nuttikung Are you still seeing this with the latest release (v0.8.4)?

**Comment by msach22** [comment]
@nuttikung I'm going to go ahead and close this issue, please re-open if this still persist with the latest release.

**Comment by smakosh** [comment]
Hi @msach22 I'm still facing the same issue above, I'm using the last version `0.8.3`

**Comment by msach22** [comment]
@smakosh Are you still seeing this with v0.9.2?

**Comment by smakosh** [comment]
let me try to upgrade to that version

**Comment by smakosh** [comment]
oh yes, still the same issue even after updating to the latest version `"opentok-react-native": "^0.9.2"`

**Comment by msach22** [comment]
Hmm, I just created a fresh react native project and added this library and am not having any issues building the application. Please make sure that you've created a Bridging Header, downloaded the OpenTok iOS SDK using CocoaPods without the `use_frameworks!` flag, and open the `.xcworkspace` file.

**Comment by smakosh** [comment]
after a fresh reset, everything worked fine, thanks

**Comment by cassianomon** [comment]
> after a fresh reset, everything worked fine, thanks

How did you fresh reset?

**Comment by msach22** [comment]
@cassianomon I think by "fresh reset' - they mean creating a new application from scratch using `react-native init`.

**Comment by cassianomon** [comment]
Oh I see, I am having the same error but with another package (react-native-facebook-login), not possible to go for a fresh start...
https://github.com/magus/react-native-facebook-login/issues/331

**Comment by msach22** [comment]
@cassianomon Ah, yea I'm not sure how it would work with the facebook login library - this is simply for this library.

**Comment by smakosh** [comment]
I upgraded to the latest RN manually & what I meant by a fresh start is that I had to install the pods, create the bridge header etc...

**Comment by cassianomon** [comment]
> I upgraded to the latest RN manually & what I meant by a fresh start is that I had to install the pods, create the bridge header etc...

Ok Thanks!

**Comment by smakosh** [comment]
So the issue is related to cocapods pre-release version that doesn't generate opentok headers inside the pod/headers folder

**Comment by misterpoloy** [comment]
This issue still exists, I'm using Xcode `10.1`
`pod 'OpenTok', '2.15.2'`
`"react-native": "0.57.4",` and 
`"opentok-react-native": "0.9.4",`

Yes, I follow all the link and installation like [this](https://github.com/opentok/opentok-react-native/issues/155) 
![Screen Shot 2019-03-13 at 2 48 52 PM](https://user-images.githubusercontent.com/22797958/54313516-24bd1980-459f-11e9-9da1-89ecab3af96f.png)

**Comment by andikare** [comment]
i have Bug like that , can anyone help me ? i still don't understand thank you

**Comment by smakosh** [comment]
1- Revert your cocoapods version back to an older version
2- Run `pod install` once again
3- Check inside `Pods/Headers/Private` and `..../Public` folders if there is the opentok folder that contains the pre-compiled headers to copy them, otherwise, download them from [here](https://drive.google.com/open?id=1WoPWubE8L6jKy49JDvTCJTZrJWF2aYYO)
Everything should be fixed

**Comment by andikare** [comment]
@smakosh  thank you for fast respond sir , sorry but i using react native android , you have solution ?

**Comment by smakosh** [comment]
Works fine for Android as long as you have followed the installation instructions on the readme file

**Comment by andikare** [comment]
@smakosh @msach22  , here is my build grade, i followed but still bug

buildscript {
    ext {
        buildToolsVersion = "27.0.3"
        minSdkVersion = 16
        compileSdkVersion = 27
        targetSdkVersion = 26
        supportLibVersion = "27.1.1"
    }
    repositories {
        jcenter()
        google()
    }
    dependencies {
        classpath 'com.android.tools.build:gradle:3.1.4'

        // NOTE: Do not place your application dependencies here; they belong
        // in the individual module build.gradle files
    }
}

allprojects {
    repositories {
        mavenLocal()
        jcenter()
        maven {
            // All of React Native (JS, Obj-C sources, Android binaries) is installed from npm
            url "$rootDir/../node_modules/react-native/android"
        }
                maven { url "https://jitpack.io" }
        maven { url "https://maven.google.com" }
        google()
         maven {
            url "http://tokbox.bintray.com/maven"
        }
    }
}


task wrapper(type: Wrapper) {
    gradleVersion = '4.4'
    distributionUrl = distributionUrl.replace("bin", "all")
}



and its my app build gradle
android {
    compileSdkVersion rootProject.ext.compileSdkVersion
    buildToolsVersion rootProject.ext.buildToolsVersion

    defaultConfig {
        applicationId "io.market.nativebase.geekyants.nativebasekitchensink"
        minSdkVersion rootProject.ext.minSdkVersion
        targetSdkVersion rootProject.ext.targetSdkVersion
        versionCode 40
        versionName "2.4.0"
        ndk {
            abiFilters "armeabi-v7a", "x86"
        }
    }
    signingConfigs {
        release {
            if (project.hasProperty('MYAPP_RELEASE_STORE_FILE')) {
                storeFile file(MYAPP_RELEASE_STORE_FILE)
                storePassword MYAPP_RELEASE_STORE_PASSWORD
                keyAlias MYAPP_RELEASE_KEY_ALIAS
                keyPassword MYAPP_RELEASE_KEY_PASSWORD
            }
        }
    }
    splits {
        abi {
            reset()
            enable enableSeparateBuildPerCPUArchitecture
            universalApk false  // If true, also generate a universal APK
            include "armeabi-v7a", "x86"
        }
    }
    buildTypes {
        release {
            minifyEnabled enableProguardInReleaseBuilds
            proguardFiles getDefaultProguardFile("proguard-android.txt"), "proguard-rules.pro"
            signingConfig signingConfigs.release
        }
    }
    // applicationVariants are e.g. debug, release
    applicationVariants.all { variant ->
        variant.outputs.each { output ->
            // For each separate APK per architecture, set a unique version code as described here:
            // http://tools.android.com/tech-docs/new-build-system/user-guide/apk-splits
            def versionCodes = ["armeabi-v7a":1, "x86":2]
            def abi = output.getFilter(OutputFile.ABI)
            if (abi != null) {  // null for the universal-debug, universal-release variants
                output.versionCodeOverride =
                        versionCodes.get(abi) * 1048576 + defaultConfig.versionCode
            }
        }
    }
}

dependencies {
        compile project(':opentok-react-native')

compile 'com.opentok.android:opentok-android-sdk:2.15.3'   

 compile project(':react-native-camera')
    compile project(':react-native-opentok')
    compile project(':react-native-fs')
    compile project(':react-native-vector-icons')
    implementation fileTree(dir: "libs", include: ["*.jar"])
    implementation "com.android.support:appcompat-v7:${rootProject.ext.supportLibVersion}"
    implementation "com.facebook.react:react-native:+"  // From node_modules
    
    //untuk gif
    implementation 'com.facebook.fresco:fresco:1.10.0'
    implementation 'com.facebook.fresco:animated-gif:1.10.0'
}

// Run this once to be able to run the application with BUCK
// puts all compile dependencies into folder libs for BUCK to use
task copyDownloadableDepsToLibs(type: Copy) {
    from configurations.compile
    into 'libs'
}

**Comment by andikare** [comment]
@nuttikung can't you solve this  ? thanks you

**Comment by kibponce** [comment]
Up

**Comment by ibrahimeraslan** [comment]
ş

**Comment by Remig9** [comment]
i am having the same issue on android when building the apk 
pls help me out

**Comment by i-suraj007** [comment]
> i am having the same issue on android when building the apk
> pls help me out

same issue +1

**Comment by HarshitPKing** [comment]
Current behavior

I have tried to compile and build for production apk and I got an error shown as
"FAILURE: Build failed with an exception.

What went wrong:
Execution failed for task ':opentok-react-native:verifyReleaseResources'.
com.android.ide.common.process.ProcessException: Failed to execute aapt "

Steps to reproduce

bundle js via react native command.
cd android && ./gradlew assembleRelease
the error will be shown.
Example Project

What is the current bug behavior?

It's shown as BUILD FAILED with error as I mention above.

What is the expected correct behavior?

It should show as BUILD SUCCESS and show up for apk file.
